lte/gateway/python/magma/pipelined/ng_manager/session_state_manager_util.py

```diff
@@ -0,0 +1,64 @@
+"""
+Copyright 2020 The Magma Authors.
+
+This source code is licensed under the BSD-style license found in the
+LICENSE file in the root directory of this source tree.
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+"""
+
+from collections import namedtuple
+
+FARRuleEntry =  namedtuple(
+                   'FARRuleEntry',
+                    ['sess_ver_no', 'apply_action', 'o_teid', 'enodeb_ip_addr'])
+
+PDRRuleEntry = namedtuple(
+                'PDRRuleEntry',
+                ['pdr_id', 'sess_ver_no', 'precedence','local_f_teid', 'ue_ip_addr',
+                 'o_h_remo_descr', 'far_action'])
+
+# Create the Named tuple for the FAR entry
+def far_create_rule_entry (far_entry) -> FARRuleEntry:
+    o_teid = None
+    fwd_enodeb_ip_addr = None
+
+    if far_entry.fwd_parm.HasField('outr_head_cr'):
+        o_teid = far_entry.fwd_parm.outr_head_cr.o_teid
+        fwd_enodeb_ip_addr = far_entry.fwd_parm.outr_head_cr.ipv4_adr
+
+
+    far_rule = FARRuleEntry (far_entry.sess_ver_no,
+                             far_entry.apply_action.apply_action[0],
+                             o_teid, fwd_enodeb_ip_addr)
+
+    return far_rule
+
+# Create the Named tuple for the PDR entry
+def pdr_create_rule_entry (pdr_entry, far_entry=None) -> PDRRuleEntry:
+    local_f_teid = None
+    ue_ip_addr = None
+    o_h_remo_descr = None
+
+    # get local teid
+    if pdr_entry.pdi.local_f_teid:
+        local_f_teid = pdr_entry.pdi.local_f_teid
+
+    # Get UE IP address
+    if len(pdr_entry.pdi.ue_ip_adr):
```

Comment:
If its a string in python lets just do `if pdr_entry.pdi.ue_ip_adr:` empty string will eval to false

Reply:
It was part of the Protobuf/GRPC message. Default value of string is "".. Hence checking the length.